src/ui/index.js

```diff
@@ -3,6 +3,13 @@ import ReactDOM from 'react-dom';
 import App from './App';
 
 const out = document.createElement('div');
+
+// Creates a unique ID
+// Namespaces the event emitter to prevent cross tab or window events
+// This is defined here so that both the UI and Preview iframe have access to it
+out.id = 'sync_id';
+out.setAttribute('data-id', new Date().getTime());
```

Comment:
I ran out of ideas, not sure if there's a better way to share a string between across an iframe